test/junit/scala/collection/ViewTest.scala

```diff
@@ -113,4 +113,11 @@ class ViewTest {
   def _toString(): Unit = {
     assertEquals("View(<not computed>)", View(1, 2, 3).toString)
   }
+
+  // see https://github.com/scala/scala/pull/9388/files#r630681050
+  @Test
+  def patch(): Unit = {
+    val view = List(2).view.patch(1, List(3, 4, 5).iterator, 0)
+    assertEquals(view.toList, view.toList)
```

Comment:
can we also assert what the value should be?

Also, could we test a patch law, something like:

```scala
forAll { (sq: SeqView[Int], from: Int, items: IterableOnce[Int], r: Int) =>
  val patchedView = Try(sq.patch(from, items, r)).toOption

  val viaList = Try {
    val (left, right) = sq.toList.splitAt(from)
    (left ++ items ++ right.drop(r)).toList
  }.toOption
 
  val asList = patchedView.map(_.toList)
  (asList == viaList) &&
    // make sure the view doesn't change
    (patchedView.map(_.toList) == viaList) 
}
```

Reply:
that code doesn't work, because `items` cannot be reused. I've added an assertion as to the actual value, however

Reply:
yes, sorry. Make `items` a `List[Int]` and then pass a `.iterator` of it as needed.

Reply:
`patch` actually doesn't care if `from` is past the end of the original collection, so `Try` wasn't needed. also, using `Try#toOption` would have likely:

- not worked because if exactly one of the `Try`s fails, the whole things fails with no information.
- hidden the bug that cropped up.

also, there's no `Arbitrary` for `SeqView`, and you end up needing to use specifically a `Seq` that doesn't extend `StrictOptimisedSeqOps` to have a meaningful test. hopefully it doesn't get changed in a future version and make the test worthless.

Reply:
regardless, I've tweaked it to work